lib/backend/etcdbk/etcd.go

```diff
@@ -649,6 +649,60 @@ func (b *EtcdBackend) NewWatcher(ctx context.Context, watch backend.Watch) (back
 	return b.buf.NewWatcher(ctx, watch)
 }
 
+func (b *EtcdBackend) Iterate(ctx context.Context, startKey, endKey backend.Key, limit int, order backend.IterationOrder) (iter.Seq2[backend.Item, error], error) {
+	if startKey.IsZero() {
+		return nil, trace.BadParameter("missing parameter startKey")
+	}
+	if endKey.IsZero() {
+		return nil, trace.BadParameter("missing parameter endKey")
+	}
+
+	sort := clientv3.SortAscend
+	if order == backend.IterateDescending {
+		sort = clientv3.SortDescend
+	}
+
+	return func(yield func(backend.Item, error) bool) {
+		// etcd's range query includes the start point and excludes the end point,
+		// but Backend.GetRange is supposed to be inclusive at both ends, so we
+		// query until the very next key in lexicographic order (i.e., the same key
+		// followed by a 0 byte)
+		opts := []clientv3.OpOption{
+			clientv3.WithRange(b.prependPrefix(endKey) + "\x00"),
+			clientv3.WithSort(clientv3.SortByKey, sort),
+		}
+		if limit > 0 {
+			opts = append(opts, clientv3.WithLimit(int64(limit)))
+		}
+		start := b.clock.Now()
+		re, err := b.clients.Next().Get(ctx, b.prependPrefix(startKey), opts...)
```

Comment:
My intuitive expectation when using an iterative API would be that under the hood it would be reading in "chunks", not all at once, and that, if anything, a "limit" to the total items yielded by iteration would be something I'd supply by wrapping the returned iterator in a combinator.

If anything, something like `ChunkSizeHint`/`BufferSizeHint` might be more appropriate than just having a hard limit value which both serves as the indicator of the number of items to buffer, and the maximum number of items to yield.